### ISSUE TITLE
CA1019 Incorrectly Reported

### ISSUE THREAD

**Issue by TonyValenti** [dotnet/roslyn-analyzers#7442]
### Analyzer

**Diagnostic ID**: [CA1019](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1019): `CA1019: Define accessors for attribute arguments`

### Analyzer source

**SDK**: .NET 8.0
**Version**: .NET 8.0

```<LangVersion>preview</LangVersion>```

### Describe the bug
CA1019 looks for a property that matches the Name AND Type of the member.  It should just look for the name and ignore the type.

### Steps To Reproduce
```
    [AttributeUsage(validOn: AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
    public class DynamicallyIncludesAttribute : Attribute {

        public ImmutableArray<Type> Types { get; }

        public DynamicallyIncludesAttribute(params IEnumerable<Type> Types) {
            this.Types = [.. Types];
        }
    }
```

### Expected behavior
CA1019 is not reported in the above code.

### Actual behavior
CA1019 is reported in the above code.
